test/node/bson_test.js

```diff
@@ -2323,4 +2323,37 @@ describe('BSON', function() {
       expect(() => bson.serialize(doc, { checkKeys: true })).to.not.throw()
     });
   });
-});
+
+  describe('UUID serialize', function() {
+    const BSON = Object.assign({}, createBSON());
+    BSON.UUID = require('bson').UUID;
+
+    const BSON_DATA_BINARY = 5;
+    const BSON_BINARY_SUBTYPE_UUID_NEW = 4;
+
+      it('should serialize BSON.UUID() input the same as BSON.UUID().toBinary()', () => {
+        const exampleUUID = new BSON.UUID();
+        const toBinarySerialization = BSON.serialize({ uuid: exampleUUID.toBinary() });
+        const plainUUIDSerialization = BSON.serialize({ uuid: exampleUUID });
+        expect(plainUUIDSerialization).to.deep.equal(toBinarySerialization);
+      });
+  
+      it('should have a valid UUID _bsontype with Object input without error', () => {
+        const output = BSON.serialize({ uuid: new BSON.UUID() });
```

Comment:
Could we also deserialize in these 3 tests and check the equality on the original UUID? I'd feel a bit more comfortable knowing that more than just the right bytes are in the right location even though there are tests in this file that do round trip testing on binaries.

Reply:
yes of course!

Reply:
Of course! The tests aren't passing when I do this, so I'll look into it more later today!

Reply:
Looks like an extra parenthesis: `expect(BSON.deserialize(plainUUIDSerialization))).to.equal(exampleUUID);` should be `expect(BSON.deserialize(plainUUIDSerialization)).to.equal(exampleUUID);`

Reply:
Fixed.